src/core/thread/csl_tx_scheduler.cpp

```diff
@@ -249,8 +249,9 @@ void CslTxScheduler::HandleSentFrame(const Mac::TxFrame &aFrame, Error aError, C
         break;
 
     case kErrorNoAck:
-        aChild.IncrementCslTxAttempts();
+        OT_ASSERT(aFrame.GetSecurityEnabled() == aFrame.IsHeaderUpdated());
```

Comment:
How about relaxing the check as this?
```suggestion
        OT_ASSERT(!aFrame.GetSecurityEnabled() || aFrame.IsHeaderUpdated());
```

In the context here, if frame security is not enabled, we  don't care about `IsHeaderUpdate()`. 

The reason I am considering for relaxing this is we may end up using `IsHeaderUpdated` for CSL IE  with MLE frame in future (which would not then use link security).

Reply:
Done.